### ISSUE TITLE
Unify test and main build system

### ISSUE THREAD

**Issue by BlakeFreer** [macformula/racecar#84]
firmware/tests uses its own build systems since it appears that googletest cannot be built with the CMake `Unix Makefiles` generator that is required for `stm`.

find a way around this bug and built `test` as a target in `firmware/CMakeLists.txt` (may need to use Docker or similar)

Once this is done, move `firmware/tests/googletest` to `firmware/third-party/googletest`

**Comment by BlakeFreer** [comment]
Do we need tests besides the `shared/util/mappers`? Is it even worth pursuing? I think most of our tests are system-level and will be incorporated in the SIL

**Comment by BlakeFreer** [comment]
> Do we need tests besides the `shared/util/mappers`? Is it even worth pursuing? I think most of our tests are system-level and will be incorporated in the SIL

Would be nice to have tests for the control system
